wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/UnsupportedEntityIdValueImpl.java

```diff
@@ -0,0 +1,160 @@
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.wikidata.wdtk.datamodel.helpers.Equality;
+import org.wikidata.wdtk.datamodel.helpers.Hash;
+import org.wikidata.wdtk.datamodel.helpers.ToString;
+import org.wikidata.wdtk.datamodel.interfaces.UnsupportedEntityIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.ValueVisitor;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonDeserializer.None;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+/**
+ * Represents a entity id value of an unsupported type.
+ * We can still "deserialize" it by just storing its
+ * JSON representation, so that it can be serialized
+ * back to its original representation.
+ * This avoids parsing failures on documents containing
+ * these values.
+ * 
+ * @author Antonin Delpeuch
+ */
+@JsonDeserialize(using = None.class)
+public class UnsupportedEntityIdValueImpl extends ValueImpl implements UnsupportedEntityIdValue {
+	
+	private final JacksonIdValue value;
+	private final String siteIri;
+
+	@JsonCreator
+	private UnsupportedEntityIdValueImpl(
+			@JsonProperty("value")
+			JacksonIdValue value,
+			@JacksonInject("siteIri")
+			String siteIri) {
+		super(JSON_VALUE_TYPE_ENTITY_ID);
+		this.value = value;
+		this.siteIri = siteIri;
+	}
+
+	private static class JacksonIdValue {
+		private final String entityType;
+		private final String id;
+		private final Map<String, JsonNode> contents;
+		
+		@JsonCreator
+		private JacksonIdValue(
+				@JsonProperty("entity-type")
+				String entityType,
+				@JsonProperty("id")
+				String id) {
+			this.entityType = entityType;
+			this.id = id;
+			contents = new HashMap<>();
+		}
+		
+		@JsonProperty("entity-type")
+		public String getEntityTypeString() {
+			return entityType;
+		}
+		
+		@JsonProperty("id")
+		public String getId() {
+			return id;
+		}
+		
+		@JsonAnyGetter
+		protected Map<String, JsonNode> getContents() {
+			return contents;
+		}
+		
+		@JsonAnySetter
+		protected void loadContents(String key, JsonNode value) {
+			this.contents.put(key, value);
+		}
+	}
+	
+	@Override
+	public String toString() {
+		return ToString.toString(this);
+	}
+
+	@Override
+	@JsonIgnore
+	public String getEntityType() {
+		return ET_UNSUPPORTED;
```

Comment:
It might be relevant to return null or something build from entityType

Reply:
Yeah, I was not too sure. In general I am not a big fan of making up recipes to build these sort of URIs, unless the rules for that are documented somewhere? Returning `null` could also cause NPEs potentially…

Reply:
Ok! Make sense!

Reply:
Sorry to change my mind. Entity types URIs are always following the pattern `"http://wikiba.se/ontology#Wikibase" + stirngEntityType.toCamlCase()`. It is exactly the same conversion that is implemented in `DatatypeIdImpl`.

Reply:
Fine with me!